maven-artifact/src/main/java/org/apache/maven/artifact/handler/ArtifactHandler.java

```diff
@@ -53,5 +53,16 @@ public interface ArtifactHandler {
 
     String getLanguage();
 
+    /**
+     * Specifies if the artifact contains java classes and can be added to the classpath.
+     * Whether the artifact <em>should</em> be added to the classpath depends on other
+     * dependency properties.
+     *
+     * @return if the artifact <em>can</em> be added to the class path
+     *
+     * @deprecated A value of {@code true} does not mean that the dependency <em>should</em>
+     * be placed on the classpath. See {@code JavaPathType} instead for better analysis.
+     */
+    @Deprecated
```

Comment:
IMHO whole artifactHandler/ArtifactHandlerManager should be deprecated, is there ONLY to support Maven3 plugins in Maven4, while since eee037e6763445e7901b84fce4e992ddec17df0d they are really just adapting maven4 stuff.